### DISCUSSION TITLE
M2M relationship in both objects

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [directus/directus#22974] [Feature Requests]
M2M fields currently only filter to the current junction field (by design). When the M2M field is recursive, it should use both the "current" junction and the "foreign" junction field 

---

### Discussed in https://github.com/directus/directus/discussions/14052

<div type='discussions-op-text'>

<sup>Originally posted by **keesvanbemmel** June 23, 2022</sup>
Hi!

Not sure if this is even possible, or if I'll go into endless loops or anything. But I have a self referencing M2M relationship. The articles collection had an M2M field called "related_articles" referencing the articles collection itself.

Now, when I add a related article to an article, I would also like to automatically set the relation the other way around. Is that a feature at the moment? Or should I go with a hook on the junction table?

Cheers!

Kees.</div>

**Comment by rijkvanzanten** [comment]
Linear: [ENG-224](https://linear.app/directus/issue/ENG-224/m2m-relationship-in-both-objects)

**Comment by Andi4711** [comment]
Hi Rijk, all directus developers
thx first for the great work - directus for me represents the future way, any IT solution is created!

However, there is a design flaw that already was raised in issue #2374 and is related to #14470 :
When creating a many-to-many self-referencing relation, and as the relationship is unidirectional, two fields (eg calling them A-ref-A_1 and A-ref-A_2) are created. One shows the related objects in the "source", the other seen from the "target".  For example:
Collection_A has a many to many relationship with itself, created in "advanced mode" to have both fields visible
editing Collection_A.Object_1 and link it with Object_2 + Object_3

in the list view, 
for Object_1 the field A-ref-A_1 now correctly shows the IDs of Object_2 and Object_3
for Object_2 and Object_3 equally correctly show the ID of Object_1, but in the the field A-ref-A_2
All is correct, but it's very unhandy to have this information in two distinct fields. 
As well, this leads to the problem described in #14470 

Now in many self-referencing relations, the direction doesn't matter and I would like to see the related objects in ONE SINGLE field, not in two distinct ones. Is there a way to do this? If not, how to add this feature? 
It would be a BIDIRECTIONAL and self-referencing M2M relation 

Best
Andreas
